site/content/blog/2023-09-11-browser-recorder/index.md

```diff
@@ -0,0 +1,126 @@
+
+## Streamlining Website Testing: Introducing the ZAP Browser Recorder Project
+
+In the ever-evolving landscape of web development and cybersecurity, the importance of comprehensive testing cannot be overstated. However, testing websites often involves repetitive pre-task activities like logging in, which can be time-consuming and cumbersome. Recognizing this challenge, the ZAP Browser Recorder project sets out to revolutionize the way websites are tested by addressing the inconvenience users face during these initial steps.
+
+## The Solution
+
+Enter the ZAP Browser Recorder project - a visionary solution to this often-overlooked challenge. The project's primary goal is to streamline the website testing process, making it more efficient and user-friendly. At its core, the ZAP Browser Recorder aims to provide a seamless bridge between user interactions and ZAP's capabilities.
+
+The proposed solution is simple yet powerful: a browser extension that records user interactions while navigating a website. These interactions are then intelligently transformed into a Zest script, a human-readable format used in ZAP. This script is generated in JSON format, ensuring compatibility and ease of use with ZAP's existing infrastructure. This innovative approach eliminates the need for manual scripting, allowing users to focus solely on testing for vulnerabilities. 
+
+## The ZAP Browser Recorder Project: A Closer Look
+
+The ZAP Browser Recorder project is comprised of two main components: the browser extension and the client add on. The browser extension is responsible for recording user interactions, while the client add on facilitates communication between the browser extension and ZAP. Together, these components form a cohesive solution that streamlines the website testing process. 
+
+As of now the browser extension supports recording mainly the following types of user interactions: 
+
+1. Clicking
+2. Typing
+3. Frame Switches 
+
+Here is how the browser extension looks like:
+
+![browser-recoder](images/browser-recorder.png)
+
+## Client add-on
+
+The [client add-on](https://www.zaproxy.org/docs/desktop/addons/client/) belongs to the ZAP extensions which helps to integrate with the browser extension. The client add-on is responsible for the following tasks:
+
+* Responsible for the communication between the browser extension and the ZAP. 
+* Used by the Zest add-on to record the user interactions and generate the Zest script.
```

Comment:
I think they should be linked when first mentioned (though does not hurt linking in several places).